### DISCUSSION TITLE
defined  model property

### DISCUSSION THREAD

**Discussion by sasha172017** [adonisjs/core#1849] [Help]
How can i defined model property without use decorator @column()? Or how can i dynamic set column for model?

**Comment by DraftProducts** [comment]
Hi @sasha172017, I'm not sure I fully understood your question so if so, could you give more information?
Models are typed and are used by Lucid ORM, so if you want to query your tables, you need to define them (with decorators).

To create a column you can use migrations, that you can generate with ace commands.
_`node ace --help` will provide you all the commands._

Adonis migrations are using knex so if you need some documentation to write them:
https://knexjs.org/#Schema-Building

Hope I answered your question.

**Reply by sasha172017** [reply]
function upload(folder?: string) {
  return (target, key: string) => {
    let val = target[key];
    const getter = () => {
      return val;
    }
    const setter = (next) => {
      val = `${next}`;
    }
    Object.defineProperty(target, key, {
      get: getter,
      set: setter,
      enumerable: true,
      configurable: true,
    });
  }
} 

  @column()
@upload('folder')
  private avatar: string | null;

i want to write own decatorator for upload file, but getter and setter dont work with @column

**Reply by DraftProducts** [reply]
```ts
function upload(folder?: string) {
    return (target, key: string) => {
        let val = target[key];
        const getter = () => {
            return val;
        }
        const setter = (next) => {
            val = ${next};
        }
        Object.defineProperty(target, key, {
            get: getter,
            set: setter,
            enumerable: true,
            configurable: true,
        });
    }
}

@column()
@Upload('folder')
private avatar: string | null;

**Reply by DraftProducts** [reply]
I never created a decorrator, however, if your goal is to upload files, i think you should create methods in your controllers to do that.
There is a page about this topic in the documentation: 
 => https://preview.adonisjs.com/guides/http/file-uploads
